test/test_time_util.py

```diff
@@ -44,3 +46,230 @@ def test_str_with_microseconds_returns_float(self):
         numeric = time_util.datetime_to_epoch(dt)
         assert (int(numeric) < float(numeric))
         assert (numeric == 1371209423.1)
+
+    def test_discrete_boundaries(self):
+
+        dte = datetime.now()
```

Comment:
Yes, it is messy there, but it took me also some time to find the right test conditions, and they are specific for the different frame-types...
Anyway, if @anitagraser and @carolinux  are positive about the functionality we can do some more cleanups